src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -0,0 +1,349 @@
+    }
+
+    /**
+     * @description Queries for default Field Mappings and builds a map of DeveloperName to Field Mapping record
+     * Used to check for existance of Field Mapping record without loop
+     * @param fieldMappingSetId Id of the default Field Mapping Set
+     * @return Collection of Field Mapping DeveloperName to Field Mapping record
+     */
+    private Map<String, Data_Import_Field_Mapping__mdt> getDefaultFieldMappings(Id fieldMappingSetId) {
+        List<Data_Import_Field_Mapping__mdt> existingFieldMappings = [SELECT Id,
+                                                                            Label,
+                                                                            DeveloperName,
+                                                                            Data_Import_Field_Mapping_Set__c,
+                                                                            Required__c,
+                                                                            Source_Field_API_Name__c,
+                                                                            Target_Field_API_Name__c,
+                                                                            Target_Object_Mapping__c,
+                                                                            Is_Deleted__c
+                                                                        FROM Data_Import_Field_Mapping__mdt
+                                                                        WHERE Data_Import_Field_Mapping_Set__c = :fieldMappingSetId];
+        Map<String, Data_Import_Field_Mapping__mdt> diFieldNameToFieldMappingMap = new Map<String, Data_Import_Field_Mapping__mdt>();
+
+        for(Data_Import_Field_Mapping__mdt fieldMapping : existingFieldMappings) {
+            diFieldNameToFieldMappingMap.put(fieldMapping.Source_Field_API_Name__c, fieldMapping);
+        }
+
+        return diFieldNameToFieldMappingMap;
+    }
+
+    /** @description Default Object Mapping Set record */
+    public static Data_Import_Object_Mapping_Set__mdt defaultObjectMappingSet {
+        get {
+            if(defaultObjectMappingSet == null) {
+                return getDefaultObjectMappingSet();
+            }
+            return defaultObjectMappingSet;
+        } private set;
+    }
+
+    /** @description Default Field Mapping Set record */
+    public static Data_Import_Field_Mapping_Set__mdt defaultFieldMappingSet {
+        get {
+            if(defaultFieldMappingSet == null) {
+                return getDefaultFieldMappingSet();
+            }
+            return defaultFieldMappingSet;
+        } private set;
+    }
+
+    /** @description Queries for default Object Mapping Set, assume existence */
+    private static Data_Import_Object_Mapping_Set__mdt getDefaultObjectMappingSet() {
+        return [SELECT Id, DeveloperName, Label FROM Data_Import_Object_Mapping_Set__mdt WHERE DeveloperName = 'Default_Object_Mapping_Set'];
```

Comment:
Can we move 'Default_Field_Mapping_Set' to a class-level variable or defined constant?

Reply:
And 'Default_Object_Mapping_Set'